### PR TITLE
Fix delete parser to reject index and name together

### LINKED ISSUE #247: [PE-D][Tester B] `delete` accepts both index and name and deletes the indexed contact
## Description
The UG says `delete` can use either `index` or `name`, but not both.

## Steps to Reproduce:
1. Execute `delete 1 /n NonExistentName`

## Expected Result
Command is rejected and exception to be thrown because both index and name were supplied.

## Actual Result
The first displayed contact is deleted.

## Screenshot
In `User.Guide.-.HitList.pdf` 

Image

1. Executing `delete 1 /n nonexistance` would delete the first contact.

Image

-------------
Labels: `severity.Low` `type.FunctionalityBug`

### PR DESCRIPTION
**Pull request by Leiruz**
Fixes #247

## Summary
- reject delete commands that provide both an index and /n
- preserve valid index-only delete parsing
- preserve valid name-only delete parsing
- add parser coverage for the combined-input bug

## Testing
- .\gradlew.bat test --tests "hitlist.logic.parser.DeleteCommandParserTest"
- .\gradlew.bat test

### CODE REVIEW

**Comment by Egg-Fish** [comment]
Seems that test is missing for a command such as `delete 123 /n` (nonempty preamble, empty prefix). Could you try adding tests for this?

File: src/main/java/hitlist/logic/parser/DeleteCommandParser.java [hunk 1]
```diff
@@ -18,27 +18,31 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the DeleteCommand
      * and returns a DeleteCommand object for execution.
+     *
```
Comment (Egg-Fish) [inline src/main/java/hitlist/logic/parser/DeleteCommandParser.java:21]:
```suggestion
```
Be careful with line count for feature freeze

Comment (Leiruz) [reply src/main/java/hitlist/logic/parser/DeleteCommandParser.java:21]: Thanks, I added a parser test for `delete 123 /n` to cover the non-empty preamble + empty name prefix case, and I also cleaned up the unnecessary line to keep the diff minimal for feature freeze.

**Comment by Egg-Fish** [comment]
Not sure why Codecov is failing in GH, though it shows 0 missed lines in the Codecov report @AY2526S2-CS2103T-W11-2/developers

File: src/main/java/hitlist/logic/parser/DeleteCommandParser.java [hunk 2]
```diff
@@ -18,27 +18,31 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
...
                 return new DeleteCommand(index);
-            } else {
-                // Preamble is not a valid index, treat as name
-                Name name = ParserUtil.parseName(nameValue.orElseThrow(() -> new ParseException(
-                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE))));
-                return new DeleteCommand(name);
             }
+
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
```
Comment (lukeetann) [inline src/main/java/hitlist/logic/parser/DeleteCommandParser.java:42]: Shouldn't be throwing an exception only to catch it the next line, but if this has less lines changed then so be it.

**Comment by lukeetann** [comment]
> Not sure why Codecov is failing in GH, though it shows 0 missed lines in the Codecov report @AY2526S2-CS2103T-W11-2/developers

The errors from codecov are from uncovered lines that are being deleted. Should be fine to merge without affecting coverage. Re-ran codecov several times without being able to fix the issue.

**Comment by niclzy** [comment]
> > Not sure why Codecov is failing in GH, though it shows 0 missed lines in the Codecov report @AY2526S2-CS2103T-W11-2/developers
> 
> The errors from codecov are from uncovered lines that are being deleted. Should be fine to merge without affecting coverage. Re-ran codecov several times without being able to fix the issue.

Seconding this, should be fine to merge, I think all edge cases are already handled correctly

**Review by niclzy — APPROVED** [review]
lgtm, test coverage can be safely ignored for this
